pkg/http/handlers/base.go

```diff
@@ -0,0 +1,115 @@
+package handlers
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/contiamo/go-base/pkg/errors"
+	"github.com/contiamo/go-base/pkg/models"
+	"github.com/contiamo/go-base/pkg/tracing"
+)
+
+const (
+	maxRequestBodySize = 1024 * 1024 // 1MB
+)
+
+// BaseHandler contains all the base functions every handler should have
+type BaseHandler interface {
+	tracing.Tracer
+	// Read reads the body and tries to decode JSON from it to the given output type
+	Parse(r *http.Request, out interface{}) error
+	// Write writes the response serving the given object as JSON with the given status
+	Write(ctx context.Context, w http.ResponseWriter, status int, obj interface{})
+	// Error serves the proper error object based on the given error and its type
+	Error(context.Context, http.ResponseWriter, error)
+}
+
+// NewBaseHandler creates a new base HTTP handler that
+// contains shared logic among all the handlers.
```

Comment:
We should mention that it is designed for JSON based REST-ful APIs.  In particular,  `Write` always writes JSON and `Parse` only parses JSON